internal/indexnode/task.go

```diff
@@ -131,30 +137,39 @@ func (it *IndexBuildTask) PostExecute(ctx context.Context) error {
 		return err
 	}
 
-	nty := &indexpb.NotifyBuildIndexRequest{
-		Status: &commonpb.Status{
-			ErrorCode: commonpb.ErrorCode_Success,
-		},
-		IndexBuildID:   it.req.IndexBuildID,
-		NodeID:         it.nodeID,
-		IndexFilePaths: it.savePaths,
+	indexMeta := indexpb.IndexMeta{}
+	resp, err := it.etcdKV.LoadWithPrefix2(it.req.MetaPath)
+	if err != nil {
+		log.Debug("IndexService", zap.Any("load meta error with path", it.req.MetaPath))
+		log.Debug("IndexService", zap.Any("Load meta error", err))
+		return err
 	}
+	err = proto.UnmarshalText(string(resp.Kvs[0].Value), &indexMeta)
+	if err != nil {
+		log.Debug("IndexService", zap.Any("Unmarshal error", err))
+		return err
+	}
+	if indexMeta.Version > it.req.Version {
+		log.Debug("IndexNode", zap.Any("Notify build index", "This version is not the latest version"))
+		return nil
+	}
+	indexMeta.IndexFilePaths = it.savePaths
+	indexMeta.State = commonpb.IndexState_Finished
 	if it.err != nil {
-		nty.Status.ErrorCode = commonpb.ErrorCode_BuildIndexError
+		indexMeta.State = commonpb.IndexState_Failed
 	}
-
-	ctx = context.TODO()
-	resp, err := it.serviceClient.NotifyBuildIndex(ctx, nty)
+	log.Debug("IndexNode", zap.Any("MetaPath", it.req.MetaPath))
+	err = it.etcdKV.CompareVersionAndSwap(it.req.MetaPath, resp.Kvs[0].Version,
```

Comment:
Fail to update the version does not mean failure. Please consider each situation carefully.

Reply:
Yes, I have fully considered what will go wrong.